service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaUtils.java

```diff
@@ -0,0 +1,63 @@
+package bio.terra.workspace.service.petserviceaccount;
+
+import com.google.api.services.iam.v1.model.Binding;
+import com.google.api.services.iam.v1.model.Policy;
+import java.util.ArrayList;
+
+// Static class of methods shared by PetSaService and GrantService
+public class PetSaUtils {
+  public static final String SERVICE_ACCOUNT_USER_ROLE = "roles/iam.serviceAccountUser";
+
+  public static Binding getOrCreateBinding(Policy saPolicy) {
+    // Populate a binding list, if there is none in the policy
+    if (saPolicy.getBindings() == null) {
+      saPolicy.setBindings(new ArrayList<>());
+    }
+
+    // Find the binding, if any
+    for (Binding binding : saPolicy.getBindings()) {
+      if (binding.getRole().equals(SERVICE_ACCOUNT_USER_ROLE)) {
+        return binding;
+      }
+    }
+
+    // Create if it doesn't exist
+    Binding newBinding = new Binding().setRole(SERVICE_ACCOUNT_USER_ROLE);
+    saPolicy.getBindings().add(newBinding);
+    return newBinding;
+  }
+
+  /**
+   * @param saPolicy policy to modify
+   * @param member member to add
+   * @return true if member was added; false if member was already there
+   */
+  public static boolean addSaMember(Policy saPolicy, String member) {
+    Binding binding = getOrCreateBinding(saPolicy);
+    if (binding.getMembers() == null) {
```

Comment:
return false.
otherwise we will have a npe below.

Reply:
There won't be an NPE, because I populate the members array, right?
That said, it makes more sense to return false as you suggest.